src/NotifyStatusPoller/Query/Handler/GetNotifyStatusHandler.php

```diff
@@ -29,6 +32,10 @@ public function handle(GetNotifyStatus $query)
     {
         $response = $this->notifyClient->getNotification($query->getNotifyId());
 
+        if (empty($response['status'])) {
+            throw new NotifyException("Notification not found for document ID: " . $query->getDocumentId());
```

Comment:
I'm not sure about reusing the NotifyException for this; if something breaks in the previous step, `getNotification` will throw a NotifyException and when we catch it in the JobRunner we want to log a CRITICAL because something went wrong. But when the response status is empty, it's not a notify error exactly, it's a valid response so we want to catch it and log just an INFO.

So throw an exception we know won't be thrown by getNotification also, and then catch that in the JobRunner. Maybe create a new one like NotificationNotFoundException?

Reply:
Amended